product_docs/docs/pgd/5/manual/07-configure-proxies.mdx

```diff
@@ -0,0 +1,171 @@
+
+```
+sudo adduser pgdproxy
+```
+
+This user will need credentials to connect to the server.
+We will create a .pgpass file with the `proxysecret` password in it.
+Then we will lock down the `.pgpass` file so it is only accessible by its owner.
+
+```
+echo -e "*:*:*:pgdproxy:proxysecret" | sudo tee /home/pgdproxy/.pgpass
+sudo chown pgdproxy /home/pgdproxy/.pgpass
+sudo chmod 0600 /home/pgdproxy/.pgpass
+```
+
+## Configure the systemd service on each host
+
+Switch the service file from using root to using the pgdproxy user
+
+```
+sudo sed -i s/root/pgdproxy/ /usr/lib/systemd/system/pgd-proxy.service
+```
+
+Reload the systemd daemon.
+
+```
+sudo systemctl daemon-reload
+```
+
+## Create a proxy config file for each host
+
+The proxy configuration file will be slightly different for each host.
+It is a YAML file which contains a cluster object. This in turn has three
+properties:
+
+The name of the PGD cluster's top-level group (as `name`).
+An array of endpoints of databases (as `endpoints`).
+The proxy definition object with a name and endpoint (as `proxy`).
+
+The first two properties will be the same for all hosts:
+
+```
+cluster:
+  name: pgd
+  endpoints:
+    - host=host-one dbname=bdrdb port=5444
+    - host=host-two dbname=bdrdb port=5444
+    - host=host-three dbname=bdrdb port=5444
+```
+
+Note that the endpoints in this example specify port=5444.
+This is necessary for EDB Postgres Advanced Server instances.
+For EDB Postgres Extended and Community PostgreSQL, this can be omitted.
+
+The third property, `proxy`, has a `name` property and an `endpoint` property.
+The `name` property should be a name created with `bdr.create_proxy` earlier, and it will be different on each host.
+The `endpoint` property is a string which defines how the proxy presents itself as a connection string.
+Conventionally, proxies should be on a different port from direct connections and typically for PGD proxies we use port 6432.
+
+```
+  proxy:
+    name: pgd-proxy-one
+    endpoint: "host=localhost dbname=bdrdb port=6432"
+```
+
+## Install a PGD proxy configuration on each host
+
+For each host, create the `/etc/edb/pgd-proxy` directory:
+
+```
+sudo mkdir -p /etc/edb/pgd-proxy
+```
+
+Then on each host, write the appropriate configuration to the `pgd-proxy-config.yml` file in the `/etc/edb/pgd-proxy` directory.
+
+For our example, this could be run on host-one to create the file.
+
+```
+cat <<EOF | sudo tee /etc/edb/pgd-proxy/pgd-proxy-config.yml
+cluster:
+  name: pgd
+  endpoints:
+    - host=host-one dbname=bdrdb port=5444
+    - host=host-two dbname=bdrdb port=5444
+    - host=host-three dbname=bdrdb port=5444
+  proxy:
+    name: pgd-proxy-one
+    endpoint: "host=localhost dbname=bdrdb port=6432"
+EOF
+```
+
+## Restart the service
+
+```
+sudo systemctl restart pgd-proxy
+```
+
+Confirm it is running correctly with:
+
+```
+sudo systemctl status pgd-proxy
+```
+
+It should show `Active: (running)` in the opening details.
+
+
```

Comment:
At this point it'd be useful to connect to a proxy instance using `psql` to show that and demonstrate how the proxy  is working. Assuming the proxy is running on all nodes in the cluster, something like the following:

---

At this point, connecting to the PGD Proxy port on any node in the cluster will result in the connection being routed to the current write lead node. For example, connecting to the proxy on `host-three` should result in the connection being routed to `node-one`:

```
$ psql -d 'host=host-three dbname=bdrdb port=6432'
postgres=# SELECT node_name FROM bdr.local_node_summary;
 node_name
-----------
 node-one
(1 row)

postgres=# SELECT node_group_name, write_lead FROM bdr.node_group_routing_summary;
 node_group_name | write_lead
-----------------+------------
 dc1             | node-one
(1 row)
```

---

(note I suppose it's possible that the write lead is not guaranteed to be `node-one` but AFAICS should be if the nodes are set up in order).

Reply:
Incorporated. Thanks.